### ISSUE TITLE
Folders

### ISSUE THREAD

**Issue by MKaidan** [satiromarra/vscode-sftp#3]
**Is this a similar or duplicate feature request?**
- [ ] I don't know. I will go check it.
- [x] No.

**Is your feature request related to a problem? Please describe.**
I have many different FTP connections (over 30), and I would like to organize them by server, essentially.

**Describe the solution you'd like**
Add a folder config option so that in the SFTP: Explorer tab I can view each connection under a folder.

So instead of 
Connection A
Connection B
Connection C
Connection D

It will show up as
Server 1
--->Connection A
--->Connection B
Server 2
--->Connection C
--->Connection D

**Describe alternatives you've considered**
I tried using profiles but I don't think that's exactly what I need. I'm also not quite sure how profiles work.

**Does this project help you?**
- [x] Yes. SFTP IS AWESOME!

Thank you!

**Comment by satiromarra** [comment]
Profiles with alternative configurations under one connection, such as an ftp connection and having two profiles, one for development and one for production.
You use DEV by default and when you want to upload to production you just change the connection.
I don't know if I have explained it well.
```json
  {
    "name": "myServer",
    "context": "project/src",
    "host": "host",
    "username": "username",
    "password": "password",
    "remotePath": "/remote/project/src",
    "profiles": {
      "dev1": {
        "name": "Server Dev",
        "context": "dev",
        "host": "dev-host",
        "remotePath": "/dev",
        "uploadOnSave": true
      },
      "prod1": {
        "name": "Server Prod",
        "context": "prod",
        "host": "prod-host",
        "remotePath": "/prod"
      }
    },
    "defaultProfile": "dev2"
  }
```

**Comment by MKaidan** [comment]
The problem with the profile configuration is that - and correct me if I'm wrong - you cannot use different usernames and passwords for each. All 4 of the connections in my example (A-D) have different usernames and passwords.

**Comment by satiromarra** [comment]
You can use other users or passwords within the profiles. 

```json
  {
    "name": "myServer",
    "context": "project/src",
    "host": "host",
    "username": "username",
    "password": "password",
    "remotePath": "/remote/project/src",
    "profiles": {
      "dev1": {
        "name": "Server Dev",
        "host": "dev-host",
      },
      "prod1": {
        "name": "Server Prod", // <- changed name for prod
        "host": "prod-host", // <- changed host for prod
        "username": "username-prod", // <- changed username for prod
        "password": "password-prod" // <- changed password for prod
      }
    },
    "defaultProfile": "dev1"
  }
```
In the visual editor you can see the fields:
<img width="655" alt="image" src="https://user-images.githubusercontent.com/1863162/221209284-3f737e35-5822-4fec-b885-39a8aa02fd88.png">
Only 1 profile can be used at a time.

**Comment by satiromarra** [comment]
@MKaidan can you test this pre-release version? https://github.com/satiromarra/vscode-sftp/releases/tag/v1.17.7-beta1
Enable in configuration, "Sftp: Group by Host"
Click on these icon to toggle between Group and Ungroup
<img width="506" src="https://user-images.githubusercontent.com/1863162/221649855-a6f0bf5e-6085-4d5d-9997-c5e42d4f5553.png">

**Comment by MKaidan** [comment]
This is perfect! I didn't even need to change my config!

**Comment by satiromarra** [comment]
@MKaidan  I have already released the final version 1.17.7, which also fixes some bugs from the beta version.
Thanks to you for your support and help to improve this extension ;)
